packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -306,19 +316,35 @@ export class TeamsBotV2Impl {
   private resolveScenarios(
     actRoles: PluginActRoles[],
     triggers: BotTrigger[],
-    hostType: HostType
+    hostType: HostType,
+    isM365: boolean | undefined
   ): string[] {
     const scenarios: string[] = [];
-    if (hostType === HostType.Function) {
-      if (actRoles.includes(PluginActRoles.Notification)) {
-        scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
-        triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+    actRoles.map((actRole) => {
+      switch (actRole) {
+        case PluginActRoles.CommandAndResponse:
+          scenarios.push(TemplateProjectsScenarios.COMMAND_AND_RESPONSE_SCENARIO_NAME);
+          break;
+        case PluginActRoles.Notification:
+          if (hostType === HostTypes.AZURE_FUNCTIONS) {
+            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
+            triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+          }
+          if (hostType === HostTypes.APP_SERVICE) {
```

Comment:
The logic looks a little weird. The scenario seems to be decided by user input directly, why we need rely on hostType? P.S. hostType is also decided by the same user input.

Reply:
Function bot needs to add an extra "function-base" template so we need `hostType` to make the judgement.